pkg/test/vpc_client/bastion.go

```diff
@@ -87,9 +88,64 @@ func (vpc *VPC) PrepareBastion(zone string) (*types.Instance, error) {
 	}
 	if len(insts) == 0 {
 		log.LogInfo("Didn't found an existing bastion, going to launch one")
-		return vpc.LaunchBastion("", zone)
+		if cidrIp == "" {
+			cidrIp = CON.RouteDestinationCidrBlock
+		}
+		_, _, err = net.ParseCIDR(cidrIp)
+		if err != nil {
+			log.LogError("CIDR IP address format is invalid")
+			return nil, err
+		}
+		userData := fmt.Sprintf("#!/bin/bash\n\t\t"+
+			"yum update -y\n\t\t"+
+			"yum install -y squid\n\t\t"+
+			"cd /etc/squid/\n\t\t"+
+			"sudo mv ./squid.conf ./squid.conf.bak\n\t\t"+
+			"sudo touch squid.conf\n\t\t"+
+			"echo \"http_port 3128\" >> /etc/squid/squid.conf\n\t\t"+
+			"echo \"acl allowed_ips src %s\" >> /etc/squid/squid.conf\n\t\t"+
+			"echo \"http_access allow allowed_ips\" >> /etc/squid/squid.conf\n\t\t"+
+			"echo \"http_access deny all\" >> /etc/squid/squid.conf\n\t\t"+
+			"systemctl start squid\n\t\t"+
+			"systemctl enable squid", cidrIp)
+
+		encodeUserData := base64.StdEncoding.EncodeToString([]byte(userData))
+		regionZone := fmt.Sprintf("%s%s", vpc.Region, zone)
+		return vpc.LaunchBastion("", regionZone, encodeUserData)
```

Comment:
May I know the final what will be the proxy address with schema and port looks like? Is there a DNS name or user:password set?

Reply:
@jameszwang Please respond the question.